tee-worker/litentry/pallets/identity-management/src/lib.rs

```diff
@@ -317,6 +326,28 @@ pub mod pallet {
 	}
 
 	impl<T: Config> Pallet<T> {
+		fn insert_identity_with_limit(
+			owner: &T::AccountId,
+			identity: &Identity,
+			context: IdentityContext<T>,
+		) -> Result<(), ()> {
+			let count = AccountIdentityContextCount::<T>::get(owner);
+			if count >= T::AccountIdentitiesLimit::get() {
+				return Err(())
+			}
+			// return Err in case of overflow
+			let new_count = count.checked_add(1).ok_or(())?;
+			AccountIdentityContextCount::<T>::insert(&owner, new_count);
+			IDGraphs::<T>::insert(&owner, &identity, context);
+			Ok(())
+		}
+
+		fn remove_identity_with_limit(owner: &T::AccountId, identity: &Identity) {
+			let count = AccountIdentityContextCount::<T>::get(owner);
```

Comment:
Maybe it's still safer to check the existence of `owner` in `AccountIdentityContextCount`, or use `try_mutate`. The reason is _theoretically_ `AccountIdentityContextCount` and `IDGraph` don't need to be always in sync - imagine we forget to modify these two maps simultaneously in later code, such check could help us to discover the problem earlier

Reply:
You mean something like this (meanwhile i've renamed few things) ?
```
		fn remove_identity_with_limit(owner: &T::AccountId, identity: &Identity) {
			let graph_len_result = IDGraphLens::<T>::try_get(owner);
			if let Ok(graph_len) = graph_len_result {
				if graph_len == 0 {
					warn!(
						"Detected IDGraphLens inconsistency, found len 0 while removing identity"
					);
					IDGraphLens::<T>::remove(owner);
				} else {
					let new_graph_len = graph_len - 1;
					if new_graph_len == 0 {
						IDGraphLens::<T>::remove(owner);
					} else {
						IDGraphLens::<T>::insert(owner, new_graph_len);
					}
				}
			} else {
				warn!("Detected IDGraphLens inconsistency, missing IdGraphLen while removing identity");
			}
			IDGraphs::<T>::remove(&owner, &identity);
		}
```